oak/common/pkcs8.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2021 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "absl/base/attributes.h"
+
+namespace oak {
+
+// Implementation of the unencrypted PKCS#8 private key encoding:
+// https://tools.ietf.org/html/rfc5208. This implementation is based on the PKCS#8 implementation in
+// the Rust ring crate (https://crates.io/crates/ring).
+namespace pkcs8 {
+
+class ByteArray {
+ public:
+  ByteArray(const std::string& in_string);
+  ByteArray(unsigned char* in_bytes, int in_len);
+  std::string ToString();
+  std::string ToBase64();
+  static ByteArray FromBase64(const std::string& in_string);
+
+  unsigned char* bytes;
+  int len;
+};
```

Comment:
Is there no other way of storing a byte array in absl on some other library? It seems odd to have to implement such a basic class, also this particular implementation seems like it would always leak memory, or did I miss where it gets released? Is it not possible to just use a `std::string` instead of this class?

Reply:
I thought it would be clearer to have byte arrays instead of strings. The other alternative is to use std::vector, but I thought that might be overkill. Replaced ByteArray with string.